wsmaster/che-core-api-auth-bitbucket/src/main/java/org/eclipse/che/security/oauth/BitbucketOAuthAuthenticator.java

```diff
@@ -11,33 +11,47 @@
  */
 package org.eclipse.che.security.oauth;
 
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import com.google.api.client.auth.oauth2.AuthorizationCodeRequestUrl;
 import com.google.api.client.util.store.MemoryDataStoreFactory;
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.HttpURLConnection;
 import java.net.URL;
+import java.util.List;
 import javax.inject.Singleton;
 import org.eclipse.che.api.auth.shared.dto.OAuthToken;
-import org.eclipse.che.commons.json.JsonHelper;
-import org.eclipse.che.commons.json.JsonParseException;
-import org.eclipse.che.security.oauth.shared.User;
 
 /** OAuth authentication for BitBucket SAAS account. */
 @Singleton
 public class BitbucketOAuthAuthenticator extends OAuthAuthenticator {
+  private final String testRequestUrl;
+  private final String bitbucketEndpoint;
+
   public BitbucketOAuthAuthenticator(
-      String clientId, String clientSecret, String[] redirectUris, String authUri, String tokenUri)
+      String bitbucketEndpoint,
+      String clientId,
+      String clientSecret,
+      String[] redirectUris,
+      String authUri,
+      String tokenUri)
       throws IOException {
+    this.bitbucketEndpoint = bitbucketEndpoint;
+    testRequestUrl =
+        bitbucketEndpoint.equals("https://bitbucket.org")
+            ? "https://api.bitbucket.org/2.0/user"
+            : bitbucketEndpoint + "/rest/api/1.0/application-properties";
```

Comment:
can we move it to a separate method and add docs, for me, it is not obvious setup

also please consider adding constants for methods

Reply:
> can we move it to a separate method and add docs, for me, it is not obvious setup

Moved to a separate method, removed the `testRequestUrl` class variable and substituted it with the method call.
> also please consider adding constants for methods

I don't think we need a constant for the bitbucket cloud url, because it is used only in a single place.